dask/array/optimization.py

```diff
@@ -115,6 +118,45 @@ def hold_keys(dsk, dependencies):
     return hold_keys
 
 
+def _is_getter_task(
+    value,
+) -> tuple[Callable, Any, Any, bool, bool | None] | None:
+    """Check if a value in a Dask graph looks like a getter.
+
+    1. Is it a tuple with the first element a known getter.
+    2. Is it a SubgraphCallable with a single element in its
+       dsk which is a known getter.
+
+    Returns the relevant getter function if it is found, otherwise returns None
+
+    TODO: the second check is a hack to allow for slice fusion between tasks produced
+    from blockwise layers and slicing operations. Once slicing operations have
+    HighLevelGraph layers which can talk to Blockwise layers this check *should* be
+    removed, and we should not have to introspect SubgraphCallables.
+    """
+    if type(value) is not tuple:
+        return None
+    first = value[0]
+    get = None
+    if first in GETTERS:
+        get = first
+    elif isinstance(first, SubgraphCallable) and len(first.dsk) == 1:
```

Comment:
Seems fine, but potentially fragile to other changes. Hopefully we can rip all of this out and replace it with highlevelgraph optimizations soon™.

Reply:
> Hopefully we can rip all of this out

:+1:

Reply:
If you only consider SubgraphCallable definitions that have a single subtask, then you might be loosing the optimization for cases where a blockwise operation is performed after the `from_array` and before the slicing (because `optimize_blockwise` will produce SubgraphCallable objects with multiple tasks).

Is it dangerous to inspect only the "outermost" function in the SubgraphCallable? Note that I am not even sure if I know what I'm talking about here. So, ignoring complex SubgraphCallable's certaily seems like a reasonable place to start.

Reply:
> Is it dangerous to inspect only the "outermost" function in the SubgraphCallable?

Thanks for taking a look at this @rjzamora. I tried this, and it does work for the simple case. However, I now think there is a much better reason why we should only allow `SubgraphCallable`s of length 1. These tasks will, in general, be rewritten in the slicing fusion. That is probably not what we want if the subgraph callable is more complex, since any inner logic to the subgraph will be thrown away.